### PR TITLE
Threads: Array

### PR DESCRIPTION
**Pull request by denisdimitrov**

### CODE REVIEW

**Comment by denisdimitrov** [comment]
Ready for review.

File: src/main/java/com/clouway/ThreadQueue.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    /**
+     * Adds element in the queue.
+     *
+     * @param data
+     */
+    public void addElement(int data) {
+        synchronized (queue) {
+            if (queue.isEmpty()) {
```
Comment (EmilGeorgiev) [inline src/main/java/com/clouway/ThreadQueue.java]: we should add element to queue until queue is full, not only when queue is empty. When queue is full thread will block and wait consumer to remove element from queue

**Comment by denisdimitrov** [comment]
Done.

File: src/main/java/com/clouway/BlockingQueue.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ *
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>
+ */
+public class BlockingQueue {
+    private final Queue<Object> queue;
+    public int size;
+
+    public BlockingQueue(int size) {
+        this.size = size;
+        this.queue = new ArrayBlockingQueue<Object>(size);
```
Comment (EmilGeorgiev) [inline src/main/java/com/clouway/BlockingQueue.java]: Use different implementation of Queue

**Pull request merged**